src/metabase/lib/js.cljs

```diff
@@ -1031,6 +1041,111 @@
          node))
      parts)))
 
+(defn ^:export string-filter-clause
+  "Creates a string filter clause based on FE-friendly filter parts. It should be possible to destructure each created
+  expression with [[string-filter-parts]]. To avoid mistakes the function requires `options` for all operators even
+  though they might not be used. Note that the FE does not support `:is-null` and `:not-null` operators with string
+  columns."
+  [operator column values options]
+  (lib.core/string-filter-clause (keyword operator)
+                                 column
+                                 (js->clj values)
+                                 (-> options js->clj (update-keys js-key->cljs-key))))
+
+(defn ^:export string-filter-parts
+  "Destructures a string filter clause created by [[string-filter-clause]]. Returns `nil` if the clause does not match
+  the expected shape. To avoid mistakes the function returns `options` for all operators even though they might not be
+  used. Note that the FE does not support `:is-null` and `:not-null` operators with string columns."
+  [a-query stage-string a-filter-clause]
+  (when-let [filter-parts (lib.core/string-filter-parts a-query stage-string a-filter-clause)]
+    (let [{:keys [operator column values options]} filter-parts]
+      #js {:operator (name operator)
+           :column   column
+           :values   (to-array (map clj->js values))
+           :options  (-> options (update-keys cljs-key->js-key) clj->js)})))
```

Comment:
Likewise this should be a named helper with the inverse name, eg. `map->obj`.

Reply:
Fixed, added `cljs-map->js-obj` and `js-obj->cljs-map`